main.go

```diff
@@ -1,14 +1,75 @@
 package main
 
 import (
-	"ava-go/bot"
+	"ava-go/handlers"
+	"errors"
 	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/bwmarrin/discordgo"
 )
 
+func start() (err error) {
+	token, ok := os.LookupEnv("BOT_TOKEN")
+	if !ok {
+		return errors.New("couldn't find environment variable $BOT_TOKEN")
+	}
+
+	goBot, err := discordgo.New("Bot " + token)
+	if err != nil {
+		return errors.New("bot auth: " + err.Error())
+	}
+
+	// declare intents (needed to be able to get member info)
+	goBot.Identify.Intents = discordgo.MakeIntent(discordgo.IntentsAll)
+	goBot.AddHandler(handlers.VoiceStateHandler)
+
+	// add slash command handlers
+	goBot.AddHandler(func(session *discordgo.Session, interaction *discordgo.InteractionCreate) {
```

Comment:
While normally in other languages Hunters suggestion of longer variable names is correct this is one area the Golang ecosystem differs from others. In go style guides shorter names are preferred for [variables](https://github.com/golang/go/wiki/CodeReviewComments#variable-names) and [receivers](https://github.com/golang/go/wiki/CodeReviewComments#receiver-names). General rule is the closer a name is used the shorter it should be. This is all just semantics though and you can do whatever you want. Another good resource for goland is [effective go](https://go.dev/doc/effective_go) or to just look at go source code, it is the best example for how to write good go.

Reply:
updated